crates/driver/src/blockchain/mod.rs

```diff
@@ -7,21 +7,12 @@ use {
 
 pub mod contracts;
 
-const MAX_BATCH_SIZE: usize = 100;
-
-#[derive(Debug, Error)]
-pub enum Error {
-    #[error("method error: {0:?}")]
-    Method(#[from] ethcontract::errors::MethodError),
-    #[error("deploy error: {0:?}")]
-    Deploy(#[from] ethcontract::errors::DeployError),
-}
-
 /// The Ethereum blockchain.
 #[derive(Debug)]
 pub struct Ethereum {
     web3: Web3<DynTransport>,
     chain_id: eth::ChainId,
+    network: eth::NetworkName,
```

Comment:
This should probably be a:
```suggestion
    network: eth::NetworkVersion,
```

Since this is not a descriptive name, but the result of a `net_version` RPC call. We can also consider moving all our code to use `eth::ChainId` exclusively instead (my personal preference, since chain ID is the newer standard to identify different networks and is specified "better" IMO).